### PR TITLE
[Doc] Add a script to combine multiple docs into a single doc

### PR DESCRIPTION
**Pull request by galiak11**
## Description 

The `combine_readme` script combines multiple documents into one. It expects a template with inlined references to documents, and inlines them into an output document.

For example, running this command will generate a `releasing.md` document in the `contributing` folder:

```bash
./scripts/combine_readme releasing contributing
```

## Issue

b/144127966
Used by: #8755

### CODE REVIEW

File: scripts/combine_readme [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	    echo "<!-- Extracted from $topic_doc_path/$file -->" >> "$expanded_topic_path"
+	    echo >> "$expanded_topic_path"
+	    cat "$file_path" >> "$expanded_topic_path"
+	    echo >> "$expanded_topic_path" # Ensure that there is always a newline between files.
+	  else
+	    echo "$line" >> "$expanded_topic_path"
+	  fi
+	done
+
+}
```
Comment (jverkoey) [inline scripts/combine_readme]: This looks like it was extracted from https://github.com/material-components/material-components-ios/blob/628b2ecb9020b0287f44b384b93946977fd07b8a/scripts/generate_readme#L101-L114, please extract it from that script to a shared `.sh` file in a separate PR first, and then both scripts can rely on the method.

**Review by jverkoey — DISMISSED** [review]

**Comment by randallli** [comment]
As stated in https://github.com/material-components/material-components-ios/pull/8755: please explain why multiple smaller readme files is better than one giant readme.

Having a script generate the final giant readme adds complexity and we should share the benefits in the PR's description so that we understand the reasoning why we did so when we look back on this commit in the future.

File: scripts/combine_readme (hunk 1 above)
Comment (galiak11) [reply scripts/combine_readme]: Done

**Review by galiak11 — COMMENTED** [review]
Request for changes has been done.

**Comment by jverkoey** [comment]
I have some additional feedback but we're heads down on a11y this week so I'd like to hold feedback / continued investment on this until next week.

**Comment by jverkoey** [comment]
Closing this PR due to inactivity, but please re-open it again if / when you intend to pursue it.

**Pull request closed without merging**